code/game/turfs/simulated/floor/lava_pool.dm

```diff
@@ -0,0 +1,57 @@
+/turf/simulated/floor/lava_pool
+		icon_state = "lava"
+		name = "lava pool"
+		desc = "A deadly pool of molten rock that incinerates almost anything on contact."
+		luminosity = 3
+		var/list/blacklist = list(/obj/item/clothing/suit/space/hardsuit/mining, /obj/item/asteroid/goliath_hide)
+
+/turf/simulated/floor/lava_pool/Entered(var/mob/living/carbon/C)
+	Incinerate(C) //To catch items before other things get processed
+	if(!istype(C))
+		return
+
+	SSobj.processing |= src
```

Comment:
it may be more efficient to add ALL of them instead of orring whenever somebody crosses the tile.
just do if(!contents.len) return in the process in that case.